tools/autograd/gen_variable_type.py

```diff
@@ -185,47 +202,74 @@ def should_trace(declaration):
     return True
 
 
-def record_trace_outputs(declaration):
-    if declaration['name'].endswith('_out'):
+def record_trace_outputs(declaration, try_outplace):
```

Comment:
nit: can we rename this to `force_outplace`? You're not really trying anything, you're just emitting a code path that does sth else